File: core/domain/src/main/java/com/gaugustini/mhfudatabase/domain/model/SearchResults.kt
package com.gaugustini.mhfudatabase.domain.model

/**
 * Represents the results of a global search across the database.
 *
 * @property armors A list of armor pieces that match the search query.
 * @property decorations A list of decorations that match the search query.
 * @property items A list of items that match the search query.
 * @property locations A list of locations that match the search query.
 * @property monsters A list of monsters that match the search query.
 * @property quests A list of quests that match the search query.
 * @property skillTrees A list of skill trees that match the search query.
 * @property skills A list of skills that match the search query.
 * @property weapons A list of weapons that match the search query.
 */
data class SearchResults(
    val armors: List<Armor> = emptyList(),
    val decorations: List<Decoration> = emptyList(),
    val items: List<Item> = emptyList(),
    val locations: List<Location> = emptyList(),
    val monsters: List<Monster> = emptyList(),
    val quests: List<Quest> = emptyList(),
    val skillTrees: List<SkillTree> = emptyList(),
    val skills: List<Skill> = emptyList(),
    val weapons: List<Weapon> = emptyList(),
)

File: lib/features/search/domain/search_results.dart
import '../../armor/domain/armor.dart';
import '../../decoration/domain/decoration.dart';
import '../../item/domain/item.dart';
import '../../location/domain/location.dart';
import '../../monster/domain/monster.dart';
import '../../quest/domain/quest.dart';
import '../../skill/domain/skill.dart';
import '../../weapon/domain/weapon.dart';

class const SearchResults({
  final List<Armor> armors = const [],
  final List<Decoration> decorations = const [],
  final List<Item> items = const [],
  final List<Location> locations = const [],
  final List<Monster> monsters = const [],
  final List<Quest> quests = const [],
  final List<SkillTree> skillTrees = const [],
  final List<Skill> skills = const [],
  final List<Weapon> weapons = const [],
}) {
  bool get isEmpty =>
      armors.isEmpty &&
      decorations.isEmpty &&
      items.isEmpty &&
      locations.isEmpty &&
      monsters.isEmpty &&
      quests.isEmpty &&
      skillTrees.isEmpty &&
      skills.isEmpty &&
      weapons.isEmpty;
}
